maas/plugins/ceph_monitoring.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+
+# Copyright 2015, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import json
+import maas_common
+import subprocess
+
+
+STATUSES = {'HEALTH_OK': 2, 'HEALTH_WARN': 1, 'HEALTH_ERR': 0}
+
+def check_command(command):
+    output = subprocess.check_output(command, stderr=subprocess.STDOUT)
+    lines = output.strip().split('\n')
+    return json.loads(lines[-1])
+
+
+def get_ceph_report(client, keyring, fmt='json'):
+    return check_command(('ceph', '--format', fmt, '--name', client, '--keyring', keyring, 'report'))
+
+
+def get_mon_statistics(report=None):
+    mon = [m for m in report['health']['health']['health_services'][0]['mons'] if m['name'] == args.host][0]
+    maas_common.metric('health', 'uint32', STATUSES[mon['health']])
+    for key, m_type in (("kb_total", "uint64"), ("kb_used", "uint64"),
+                        ("kb_avail", "uint64"), ("avail_percent", "uint32")):
+        maas_common.metric(key, m_type, mon[key])
```

Comment:
@mattt416 I didn't do that because the metrics will be associated with a mon check but I can add it in if you would prefer